service.sleepy.watchdog/default.py

```diff
@@ -0,0 +1,293 @@
+                notifyLog('user activity detected, reset idle time')
+                _msgCnt = 0
+                _maxIdleTime = self.maxIdleTime
+                _currentIdleTime = 0
+
+            _msgCnt += 1
+
+            # Check if GlobalIdle longer than maxIdle and we're in a time frame
+
+            if self.activeTimeFrame(debug=True):
+                if _currentIdleTime > (_maxIdleTime - int(self.notifyUser)*self.notificationTime):
+
+                    notifyLog('max idle time reached, ready to perform some action')
+
+                    # Check if notification is allowed
+                    if self.notifyUser:
+                        count = 0
+                        notifyLog('init notification countdown for action no. %s' % (self.action))
+
+                        while (self.notificationTime - count > 0):
+                            notifyUser(__LS__(32115) % (__LS__(self.action), self.notificationTime - count))
+                            xbmc.sleep(9000)
+                            count += 9
+                            if _currentIdleTime > xbmc.getGlobalIdleTime():
+                                self.actionCanceled = True
+                                break
+
+                    if not self.actionCanceled:
+
+                        self.sendCecCommand()
+                        {
+                        32130: self.stopVideoAudioTV,
+                        32131: self.systemReboot,
+                        32132: self.systemShutdown,
+                        32133: self.systemHibernate,
+                        32134: self.systemSuspend,
+                        32135: self.runAddon,
+                        32136: self.quit
+                        }.get(self.action)()
+                        #
+                        # ToDo: implement more user defined actions here
+                        #       Action numbers are defined in settings.xml/strings.xml
+                        #       also see LANGOFFSET
+                        #
+                        if self.testConfig:
+                            notifyLog('watchdog was running in test mode, keep it alive')
+                        else:
+                            if self.keepAlive:
+                                notifyLog('keep watchdog alive, update idletime for next cycle')
+                                _maxIdleTime += self.maxIdleTime
+                            else:
+                                break
+                    else:
+                        notifyLog('Countdown canceled by user action')
+                        notifyUser(__LS__(32118), icon=__iconError__)
+
+                    # Reset test status
+                    if self.testConfig:
+                        __addon__.setSetting('testConfig', 'false')
+
+            else:
+                notifyLog('no active timeframe yet')
+
+            #
+            _loop = 1
+            while not xbmc.Monitor.abortRequested(self):
```

Comment:
**Someone else please comment on that.**

Never seen a loop like that.
Always was under the impression you'd do something like:
```python
monitor = xbmc.Monitor()
while not monitor.abortRequested():
    # do stuff
    monitor.waitForAbort(<time in sec>)
```